x-pack/plugins/endpoint/public/embeddables/resolver/types.ts

```diff
@@ -174,9 +189,26 @@ export interface ProcessEvent {
     source_id?: number;
     process_name: string;
     process_path: string;
+    signature_status?: string;
   };
 }
 
+/**
+ * A map of Process Ids that indicate which processes are adjacent to a given process along
+ * directions in two axes: up/down and previous/next.
+ */
+export interface AdjacentProcessMap {
+  readonly self: string;
+  up: string | null;
+  down: string | null;
+  previous: string | null;
```

Comment:
Random question, do `previous` and `next` only point to nodes within a single family (all have the same parent), or do they link across the entire level? Also not sure if we need to consider weird edge cases related to that.

Reply:
previous / next point to nodes on the same level. e.g. first node will not have a `previous` and last child on level won't have a `next`